src/layout/layout.rs

```diff
@@ -1460,14 +1464,14 @@ mod tests {
             #[case(Rect::new(0, 0, 1, 1), &[TEN, NINETY], "b")]
             #[case(Rect::new(0, 0, 1, 1), &[TEN, FULL], "b")]
             #[case(Rect::new(0, 0, 1, 1), &[TEN, DOUBLE], "b")]
-            #[case(Rect::new(0, 0, 1, 1), &[HALF, ZERO], "a")]
-            #[case(Rect::new(0, 0, 1, 1), &[HALF, HALF], "a")]
-            #[case(Rect::new(0, 0, 1, 1), &[HALF, FULL], "a")]
-            #[case(Rect::new(0, 0, 1, 1), &[HALF, DOUBLE], "a")]
-            #[case(Rect::new(0, 0, 1, 1), &[NINETY, ZERO], "a")]
-            #[case(Rect::new(0, 0, 1, 1), &[NINETY, HALF], "a")]
-            #[case(Rect::new(0, 0, 1, 1), &[NINETY, FULL], "a")]
-            #[case(Rect::new(0, 0, 1, 1), &[NINETY, DOUBLE], "a")]
+            #[case(Rect::new(0, 0, 1, 1), &[HALF, ZERO], "b")]
+            #[case(Rect::new(0, 0, 1, 1), &[HALF, HALF], "b")]
+            #[case(Rect::new(0, 0, 1, 1), &[HALF, FULL], "b")]
+            #[case(Rect::new(0, 0, 1, 1), &[HALF, DOUBLE], "b")]
+            #[case(Rect::new(0, 0, 1, 1), &[NINETY, ZERO], "b")]
+            #[case(Rect::new(0, 0, 1, 1), &[NINETY, HALF], "b")]
+            #[case(Rect::new(0, 0, 1, 1), &[NINETY, FULL], "b")]
+            #[case(Rect::new(0, 0, 1, 1), &[NINETY, DOUBLE], "b")]
```

Comment:
- 0/100 is 100% wrong
- 50/50 is debatable. I it should allocate the space as 0.5, 0.5 then round left to right giving the first item the space)
- 50/100, 50/200 are probably better 
- 90/0, 90/50 are wrong for the same reason that 0/100 is